pkg/scalers/apache_kafka_scaler.go

```diff
@@ -49,41 +48,81 @@ type apacheKafkaScaler struct {
 }
 
 type apacheKafkaMetadata struct {
-	bootstrapServers       []string
-	group                  string
-	topic                  []string
-	partitionLimitation    []int32
-	lagThreshold           int64
-	activationLagThreshold int64
-	offsetResetPolicy      offsetResetPolicy
-	allowIdleConsumers     bool
-	excludePersistentLag   bool
+	BootstrapServers       []string          `keda:"name=bootstrapServers,       order=triggerMetadata;resolvedEnv"`
+	Group                  string            `keda:"name=consumerGroup,          order=triggerMetadata;resolvedEnv"`
+	Topic                  []string          `keda:"name=topic,                  order=triggerMetadata;resolvedEnv, optional"`
+	PartitionLimitation    []int             `keda:"name=partitionLimitation,    order=triggerMetadata, optional, rangeSeparator=-"`
+	LagThreshold           int64             `keda:"name=lagThreshold,           order=triggerMetadata, default=10"`
+	ActivationLagThreshold int64             `keda:"name=activationLagThreshold, order=triggerMetadata, default=0"`
+	OffsetResetPolicy      offsetResetPolicy `keda:"name=offsetResetPolicy,      order=triggerMetadata, enum=earliest;latest, default=latest"`
+	AllowIdleConsumers     bool              `keda:"name=allowIdleConsumers,     order=triggerMetadata, optional"`
+	ExcludePersistentLag   bool              `keda:"name=excludePersistentLag,   order=triggerMetadata, optional"`
 
 	// If an invalid offset is found, whether to scale to 1 (false - the default) so consumption can
 	// occur or scale to 0 (true). See discussion in https://github.com/kedacore/keda/issues/2612
-	scaleToZeroOnInvalidOffset bool
-	limitToPartitionsWithLag   bool
+	ScaleToZeroOnInvalidOffset bool `keda:"name=scaleToZeroOnInvalidOffset, order=triggerMetadata, optional"`
+	LimitToPartitionsWithLag   bool `keda:"name=limitToPartitionsWithLag,   order=triggerMetadata, optional"`
 
 	// SASL
-	saslType kafkaSaslType
-	username string
-	password string
+	SASLType kafkaSaslType `keda:"name=sasl,     order=triggerMetadata;authParams, enum=none;plaintext;scram_sha256;scram_sha512;gssapi;aws_msk_iam, default=none"`
+	Username string        `keda:"name=username, order=authParams,                 optional"`
+	Password string        `keda:"name=password, order=authParams,                 optional"`
 
 	// MSK
-	awsRegion        string
-	awsEndpoint      string
-	awsAuthorization awsutils.AuthorizationMetadata
+	AWSRegion        string `keda:"name=awsRegion,     order=triggerMetadata, optional"`
+	AWSEndpoint      string `keda:"name=awsEndpoint,   order=triggerMetadata, optional"`
+	AWSAuthorization awsutils.AuthorizationMetadata
```

Comment:
not refactoring this on purpose, it has a deprecation notice on part of the config and I think it will be a lot easier to tackle once the deprecation happens
https://github.com/kedacore/keda/blob/4fa107915db4f5aa525f663f93cceafb1e04005c/pkg/scalers/aws/aws_authorization.go#L26
https://github.com/kedacore/keda/blob/4fa107915db4f5aa525f663f93cceafb1e04005c/pkg/scalers/aws/aws_common.go#L64